opensrp-app/src/main/java/org/smartregister/util/FormUtils.java

```diff
@@ -1147,4 +1158,66 @@ public JSONObject getFormJsonFromRepositoryOrAssets(String formIdentity) {
         return getFormJson(formIdentity);
     }
 
+    @Nullable
+    public JSONObject getSubFormJsonFromRepository(String formIdentity, String subFormsLocation, Context context, boolean translateSubForm) {
+        ClientFormRepository clientFormRepository = CoreLibrary.getInstance().context().getClientFormRepository();
+        String locale = mContext.getResources().getConfiguration().locale.getLanguage();
+
+        //Check the current locale of the app to load the correct version of the form in the desired language
+        String localeFormIdentity = formIdentity;
+        if (!"en".equals(locale)) {
+            localeFormIdentity = localeFormIdentity + "-" + locale;
+        }
+
+        String dbFormName = TextUtils.isEmpty(subFormsLocation) ? localeFormIdentity : subFormsLocation + "/" + localeFormIdentity;
```

Comment:
`StringUtils.isNotBlank` or `StringUtils.isBlank` is more efficient than `TextUtils.isEmpty` since it checks for whitespaces too